routes/foodtrucks.go

```diff
@@ -149,3 +164,107 @@ func GetFoodTrucksHandler(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	json.NewEncoder(w).Encode(foodTrucks)
 }
+
+func PutFoodTrucksHandler(w http.ResponseWriter, r *http.Request) {
+
+	// Checks for food truck ID
+	params := mux.Vars(r)
+	foodTruckID, foodTruckIDExists := params["foodTruckID"]
+	if !foodTruckIDExists {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Get user from context
+	_, userLoggedIn := r.Context().Value(middleware.UserKey).(string)
+
+	// Check for a user
+	if !userLoggedIn {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	foodTruckDecoder := json.NewDecoder(r.Body)
+	foodTruckDecoder.DisallowUnknownFields()
+
+	// Decode request
+	var currentFoodTruck updateFoodTruckRequest
+	err := foodTruckDecoder.Decode(&currentFoodTruck)
+	if err != nil {
+		log.Printf("ERROR: %v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Set tags to an empty array if they don't exist
+	tags := currentFoodTruck.Tags
```

Comment:
wait why do we do this for tags but not photos?

Reply:
idk i looked at ur code from POST and for photos it didnt have anything. will look at it rn

Reply:
wait but if location is never nil, then tags is never nil right?